crates/nodeapi/src/lib.rs

```diff
@@ -64,14 +63,83 @@ impl NodeApi {
         }
     }
 
+    /// Add a service to the node
+    pub fn register_service(
```

Comment:
Is there a reason to keep this as `pub`? Same for `update_service_status` and `deregister_service`

Reply:
done